x/emissions/abci.go

```diff
@@ -146,3 +146,40 @@ func DistributeTssRewards(ctx sdk.Context, amount sdk.Int, bankKeeper types.Bank
 	coin := sdk.NewCoins(sdk.NewCoin(config.BaseDenom, amount))
 	return bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, types.UndistributedTssRewardsPool, coin)
 }
+
+func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	allWithdrawEmissions := keeper.GetAllWithdrawEmissions(ctx)
+	// delete all withdraw emissions irrespective of weather they are processed or not; this is to avoid processing the same withdraw emission again if in case it fails
+	// if a witdraw emission fails, it will need to be created again by the observer
+	defer func() {
+		for _, withdrawEmission := range allWithdrawEmissions {
+			keeper.DeleteWithdrawEmissions(ctx, withdrawEmission.Address)
+		}
+	}()
```

Comment:
Why using a defer here? It seems we never return an error in the function body

Reply:
I used it just for code readability.